app.js

```diff
@@ -1,38 +1,65 @@
 var http = require('http'),
+    mongoose = require('./server/db/index.js'),
     morgan = require('morgan'),
     express = require('express'),
     app = express(),
     server = http.createServer(app),
     session = require('express-session'),
     bodyParser = require('body-parser'),
+    MongoStore = require('connect-mongo')(session),
+    csrf = require('csurf'),
+    csrfProtection = csrf(),
     adminRouter = require('./server/routes/admin.js'),
     apiRouter = require('./server/routes/api.js'),
-    indexRouter = require('./server/routes/index.js');
-
-if (app.get('env') !== 'production') {
-    var webpack = require('webpack'),
-    webpackDevMiddleware = require('webpack-dev-middleware'),
-    webpackHotMiddleware = require('webpack-hot-middleware'),
-    webpackConfig = require('./webpack.dev.config'),
-    webpackCompiler = webpack(webpackConfig);
-}
+    indexRouter = require('./server/routes/index.js'),
+    config = require('./config/default.js');
 
+// Logging
 morgan.token('remote-addr', function(req, res) {
     return req.headers['x-forwarded-for'] || req.ip;
 });
+app.use(morgan('combined'));
 
-app.use(morgan("combined"));
-
+// Request parsers
 app.use(bodyParser.urlencoded({
     extended: true
 }));
 app.use(bodyParser.json());
 
+// Server side views (if used)
 app.set('view engine', 'pug');
 app.set('views', './server/views');
 app.set('view cache', false);
 
-if (app.get('env') !== 'production') {
+// Initialize session
+var sessionStore = new MongoStore({
+    url: 'mongodb://' + config.mongo_hostname + '/' + config.sessions_db
+});
+
+app.use(session({
+    secret: config.secret,
+    store: sessionStore,
+    resave: false,
+    saveUninitialized: false
+}));
+
+// Set an xsrf-token for the session if it's enabled
+app.use(function(req, res, next) {
+    if (req.csrfToken) {
+        res.cookie('xsrf-token', req.csrfToken());
+    }
+
+    return next();
+});
+
+// Intiialize development webpack (hot reloading, etc);
+if (app.get('env') !== 'production' && !process.env.APIWORK) {
```

Comment:
What is the APIWORK environment doing? I assume for when working on just the backend API?

Also do we want to add this to config with default of false?